packages/gestalt/src/TableOfContents/TableOfContentsAnchor.js

```diff
@@ -0,0 +1,57 @@
+// @flow strict
+import { type Node, useState } from 'react';
+import classNames from 'classnames';
+import styles from './TableOfContentsAnchor.css';
+import Box from '../Box.js';
+import Colors from '../Colors.css';
+import Flex from '../Flex.js';
+import Layout from '../Layout.css';
+import TapArea from '../TapArea.js';
+import Text from '../Text.js';
+
+type Props = {|
+  label: string,
+  href: string,
+  active?: boolean,
+  nested?: boolean,
+  onClick?: () => void,
+|};
+
+export default function TableOfContentsAnchor({
+  label,
+  active,
+  href,
+  nested,
+  onClick,
+}: Props): Node {
+  const [focused, setFocused] = useState(false);
```

Comment:
We have a hook for this called `useInteractionStates` I believe, and you can pass the methods to tap area